src/main/java/org/terasology/logic/health/DamageAuthoritySystem.java

```diff
@@ -236,4 +236,29 @@ public void onCrash(HorizontalCollisionEvent event, EntityRef entity, CharacterS
             }
         }
     }
+    /**
+     * Reduces the baseDamage of BeforeDamagedEvent if DamageResistComponent is added.
+     *
+     * @param event BeforeDamagedEvent sent before inflicting damage
+     * @param entity Entity which suffered some type of damage
+     */
+    @ReceiveEvent
+    public void onResistMode(BeforeDamagedEvent event, EntityRef entity, DamageResistComponent resistanceComponent) {
+        String damageType = event.getDamageType().getName();
+        //takes the damage type name from the prefab name
+        String subString = damageType.substring(damageType.indexOf(':') + 1);
+        float amount;
```

Comment:
minor style issue: Not necessary to declare variable at this place - looks , you can write data in variable at more then one place